### DISCUSSION TITLE
stddef.h can't be auto inserted

### DISCUSSION THREAD

**Discussion by SBZORRO** [clangd/clangd#1835] [Q&A] [answered]
`siz` can be autocompleted to `size_t`, but it shows 

> clang [unknown_typename]: Unknown type name 'size_t'

I manually include `stdd`, it autocompleted to `stddef.h`, and shows 

> /usr/lib64/clang/17/include/stddef.h
> provides size_t

It seems that `stddef.h` can be found by clangd, but I don't understand why `stddef.h` can't be auto inserted.
Any idea?

I user emacs + eglot + clangd.
Here is the log I got from `M-x eglot-events-buffer`
[log.txt](https://github.com/clangd/clangd/files/13466906/log.txt)

**Comment by HighCommander4** [comment, marked as answer]
I can reproduce this behaviour with clangd 17, testing in an empty C source file in a new project.

I'm using a different client (VSCode), so the issue is on the server side.

Looks like a bug to me.

**Reply by HighCommander4** [reply]
I'll add that the issue occurs even if the project has an index (requires a `compile_commands.json`), and an existing source file includes `<stddef.h>` (so that clangd should be able to find the `size_t` symbol in the project's index based on regular indexing of source files, rather than having to rely on the feature to [eagerly index](https://clangd.llvm.org/config.html#standardlibrary) standard library headers).

**Reply by HighCommander4** [reply]
See also https://github.com/clangd/clangd/issues/376.

[CSymbolMap.inc](https://searchfox.org/llvm/rev/9ba74c2ee044381544c8e1e736c4c85a1a9d56bf/clang/lib/Tooling/Inclusions/Stdlib/CSymbolMap.inc) still doesn't contain an entry for `size_t`.
